yt_dlp/extractor/instagram.py

```diff
@@ -42,13 +43,15 @@ class InstagramBaseIE(InfoExtractor):
 
     _API_BASE_URL = 'https://i.instagram.com/api/v1'
     _LOGIN_URL = 'https://www.instagram.com/accounts/login'
-    _API_HEADERS = {
-        'X-IG-App-ID': '936619743392459',
-        'X-ASBD-ID': '198387',
-        'X-IG-WWW-Claim': '0',
-        'Origin': 'https://www.instagram.com',
-        'Accept': '*/*',
-    }
+
+    def _api_headers(self):
+        return {
+            'X-IG-App-ID': self._configuration_arg('app_id', '936619743392459')[0],
```

Comment:
- default value needs to be a list with 1 string element or else we are slicing the string
- this is in the baseclass, so let's lock in the `ie_key` (or else we'd have to document `instagramstory:` etc)
```suggestion
            'X-IG-App-ID': self._configuration_arg('app_id', ['936619743392459'], ie_key=InstagramIE)[0],
```

Also I had a thought about keeping a map of known/common app IDs and allowing the user to pass `web`, `ios`, etc... but if we only know those 2 then I don't think it's worth it. What do you think?

Reply:
Agree, the extent, mapping, and stability of app IDs is not apparent at this point and not clear if there's a use case for overriding by anyone other than myself, so this seems sufficient for now. Perhaps some more exhaustive research could result in a future PR that does what you suggest.